src/xds/client.rs

```diff
@@ -94,10 +94,21 @@ impl Config {
             pending: Default::default(),
             demand: rx,
             demand_tx: tx,
+            node_id: generate_node_id(), 
         }
     }
 }
 
+fn generate_node_id() -> String {
+    // TODO: may add new type, but this requires changing together with pilot-discovery 
+    let node_type = "sidecar";
+    let ip = std::env::var("INSTANCE_IP").unwrap_or_else(|_| "1.1.1.1".to_string());
```

Comment:
nit : unwrap_or

Reply:
clippy lint warns about this.

[Clippy Lints](https://rust-lang.github.io/rust-clippy/master/index.html#or_fun_call)

unwrap_or_else is more suitable for  default is a function call.